### PR TITLE
Fix last line of date picker being unresponsive

### LINKED ISSUE #2745: Dates from last week of each month cannot be selected
## 🐛 Bug Report
When you want to edit a time entry and select a different start or end date, you are not able to select any date that is from the last week of each month. It does not matter if the current date falls into the last week of a month, you still cannot select any date from that last week.

It is also valid for previous months, not only for the current one.

This seems to be an issue for Huawei P9 Lite.

## Context
This issue was first encountered by a user (using Huawei P9 Lite) and later confirmed by me once I've tested it on the same phone.

I've used new time entries (created for this issue) as well as old time entries, same behavior.

I've also manually changed the time on my phone (to May & April) to create time entries in previous months - results were the same, you cannot select any date that is in the last week of a month.

That week is clearly shown on the actual screen, there are no clipping issues, it is just unclickable (screenshots below).
image
_June - can only select dates from 14 - 24_

image
_July - can only select dates from 1 - 22_ || July 26 is the current date and is selected by default; if I click any other date, i.e. July 19, I won't be able to click July 26 again

Build/Branch: **Toggl Giskard - 1.1** - downloaded from the Play Store
Phone OS: **Android**
Phone model: **Huawei P9 Lite**

## Expected behavior
Select any date from the calendar when editing a time entry.

## To reproduce
This might be related only to particular phone model(s); I just tried to select/edit a date from last week of a month and it just wouldn't let me - those dates are just not reacting.

### PR DESCRIPTION
**Pull request by dshr**
## What's this?
A little fix that helps the native date picker UI — when a month over spans 5 weeks, the last week becomes unresponsive.

### Relationships
Closes #2745

## Why do we want this?
So people can use the date picker.

## How is it done?
By adding a negative bottom margin to the date picker equal to the height of the day cell.

### Why not another way?
The date picker is a native android one, so the only other option is writing one from scratch, which is way too much effort.

### Side effects
None that I know of.

## Review hints
1. Open the Edit TE screen
2. Try changing the start date or the end date to a day that falls on a 5th week of a month.

## :squid: Permissions
Go for it!

### CODE REVIEW

**Comment by daividssilverio** [comment]
#2950 should be merged before this. This pr's branch is broken.

**Comment by daividssilverio** [comment]
This fix was meant to make the last visible line of the calendar clickable, right?
It's working, but I can't see the last line of the calendar of December 2017 for example. Dec 31 is missing.

File: Toggl.Giskard/Resources/layout/SelectTimeFragment.axml [hunk 1]
```diff
@@ -53,4 +53,4 @@
             android:layout_height="36dp"
             android:layout_marginRight="16dp" />
     </LinearLayout>
-</RelativeLayout>
\ No newline at end of file
+</RelativeLayout>
```
Comment (daividssilverio) [inline Toggl.Giskard/Resources/layout/SelectTimeFragment.axml]: Also, discard this changes that don't change anything.

**Comment by daividssilverio** [comment]
Hey, I think you can also fix #2777 together with this one by changing the fragment dialog height.

**Comment by dshr** [comment]
Yeah, working on #2777 right now. Tweaking the height makes "shorter" months look weird, so I'm adding a stretchy pager view.

**Comment by daividssilverio** [comment]
Don't forget to assign yourself o/

**Comment by dshr** [comment]
Didn't work, the `DatePicker` always says it's the same height. I'm doing the easy thing and simply add more space.

Also closing this, since the new PR will handle it.

**Pull request closed without merging**